### PR TITLE
Remove get_data, get_values from dataset and update tests to not use it

### PR DESCRIPTION
**Pull request by jenshnielsen**
It turns out that get_parameter_data has slightly different ways of working for negative start and end. That has been fixed in this pr to match get_data as it was

### CODE REVIEW

File: qcodes/dataset/data_set.py [hunk 1]
```diff
@@ -849,47 +849,6 @@ def _validate_parameters(*params: Union[str, ParamSpec, _BaseParameter]
                 valid_param_names.append(maybeParam)
         return valid_param_names
 
-    @deprecate('This method does not accurately represent the dataset.',
-               'Use `get_parameter_data` instead.')
-    def get_data(self,
```
Comment (astafan8) [inline qcodes/dataset/data_set.py:854]: we also need to remove `get_values` and `get_setpoints`

File: qcodes/tests/dataset/test_dataset_basic.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 import re
 from unittest.mock import patch
 import random
-from typing import Sequence, Dict, Tuple, Optional
+from typing import Sequence, Dict, Tuple, Optional, List
```
Comment (astafan8) [inline qcodes/tests/dataset/test_dataset_basic.py:6]: is List used somewhere?
```suggestion
from typing import Sequence, Dict, Tuple, Optional
```

File: qcodes/dataset/sqlite/queries.py [hunk 3]
```diff
@@ -292,8 +292,8 @@ def get_parameter_tree_values(conn: ConnectionPlus,
         index is parameter value (first toplevel_param, then other_param_names)
     """
 
-    offset = (start - 1) if start is not None else 0
-    limit = (end - offset) if end is not None else -1
+    offset = max((start - 1), 0) if start is not None else 0
+    limit = max((end - offset), 0) if end is not None else -1
```
Comment (astafan8) [inline qcodes/dataset/sqlite/queries.py:296]: if my memory is not failing on me, we do have a parametrized test for the start/end, so we could add the negative cases there.

File: qcodes/tests/dataset/test_measurement_context_manager.py [hunk 4]
```diff
@@ -1457,16 +1458,17 @@ def test_datasaver_array_parameters_array(channel_array_instrument, DAC, N,
...
 
-    setpoint_arrays = ds.get_data('dummy_channel_inst_ChanA_this_setpoint')
-    data_arrays = ds.get_data('dummy_channel_inst_ChanA_dummy_array_parameter')
+    setpoint_arrays = loaded_data['dummy_channel_inst_ChanA_this_setpoint']
+    data_arrays = loaded_data['dummy_channel_inst_ChanA_dummy_array_parameter']
     assert len(setpoint_arrays) == expected_npoints
     assert len(data_arrays) == expected_npoints
 
-    data = datasaver.dataset.get_parameter_data()['dummy_channel_inst_ChanA_dummy_array_parameter']
+    # data = datasaver.dataset.get_parameter_data()['dummy_channel_inst_ChanA_dummy_array_parameter']
```
Comment (astafan8) [inline qcodes/tests/dataset/test_measurement_context_manager.py:1472]:
```suggestion
```

File: qcodes/tests/dataset/test_measurement_context_manager.py [hunk 5]
```diff
@@ -1478,18 +1480,18 @@ def test_datasaver_array_parameters_array(channel_array_instrument, DAC, N,
...
+    assert_allclose(loaded_data['dummy_dac_ch1'], expected_dac_data)
+    assert_allclose(loaded_data['dummy_channel_inst_ChanA_this_setpoint'],
                     expected_sp_data)
-    assert_allclose(data['dummy_channel_inst_ChanA_dummy_array_parameter'],
+    assert_allclose(loaded_data['dummy_channel_inst_ChanA_dummy_array_parameter'],
                     expected_output)
 
     if storage_type == 'array':
         # for now keep testing the old way of getting data (used by
         # plot_by_id). Hopefully this will eventually be deprecated
```
Comment (astafan8) [inline qcodes/tests/dataset/test_measurement_context_manager.py:1491]:
```suggestion
        # also test get_data_by_id (used by plot_by_dataset)
```

File: qcodes/tests/dataset/test_measurement_context_manager.py [hunk 6]
```diff
@@ -1554,20 +1557,20 @@ def test_datasaver_complex_array_parameters_array(channel_array_instrument,
...
+    assert data_arrays.shape == (N, M)
 
-    for data_arrays, setpoint_array in zip(data_arrays, setpoint_arrays):
-        assert_array_equal(setpoint_array[0], np.linspace(5, 9, 5))
-        assert_array_equal(data_arrays[0], np.arange(5) - 1j*np.arange(5))
+    for data_array, setpoint_array in zip(data_arrays, setpoint_arrays):
+        assert_array_equal(setpoint_array, np.linspace(5, 9, 5))
+        assert_array_equal(data_array, np.arange(5) - 1j*np.arange(5))
 
     datadicts = get_data_by_id(datasaver.run_id)
```
Comment (astafan8) [inline qcodes/tests/dataset/test_measurement_context_manager.py:1575]:
```suggestion
    # also test get_data_by_id (used by plot_by_dataset)
    datadicts = get_data_by_id(datasaver.run_id)
```

File: qcodes/tests/dataset/test_dataset_basic.py (hunk 2 above)
Comment (jenshnielsen) [reply qcodes/tests/dataset/test_dataset_basic.py:6]: Yes in test_dependent_parameters , no idea how that worked previously

File: qcodes/tests/dataset/test_measurement_context_manager.py (hunk 5 above)
Comment (jenshnielsen) [reply qcodes/tests/dataset/test_measurement_context_manager.py:1491]: We should still deprecate this one  get_data_by_id eventually but sure we can remove it from here

File: qcodes/dataset/data_set.py (hunk 1 above)
Comment (jenshnielsen) [reply qcodes/dataset/data_set.py:854]: +1 for get_values but get_setpoints is not deprecated :( It probably shuld have been

File: qcodes/dataset/sqlite/queries.py (hunk 3 above)
Comment (jenshnielsen) [reply qcodes/dataset/sqlite/queries.py:296]: Well I ported the parameterized test from get_data to get_parameter_data. That was how I discovered this. I can't find a similar test for get parameter data

Comment (astafan8) [reply qcodes/dataset/sqlite/queries.py:296]: oh, ok )

File: qcodes/dataset/data_set.py (hunk 1 above)
Comment (astafan8) [reply qcodes/dataset/data_set.py:854]: > It probably shuld have been

true. but it's ok, i guess.

File: qcodes/dataset/data_set.py [hunk 7]
```diff
@@ -1074,19 +1033,6 @@ def write_data_to_text_file(self, path: str,
...
-        Get the values (i.e. not NULLs) of the specified parameter
-        """
-        if param_name not in self.parameters:
-            raise ValueError('Unknown parameter, not in this DataSet')
-
-        values = get_values(self.conn, self.table_name, param_name)
-
-        return values
-
     def get_setpoints(self, param_name: str) -> Dict[str, List[List[Any]]]:
```
Comment (astafan8) [inline qcodes/dataset/data_set.py:1036]: shall we at last put a deprecation on get_setpoints, if we are not removing it now?

Comment (jenshnielsen) [reply qcodes/dataset/data_set.py:1036]: Yes but let me do that in a different pr. IMHO for the future deprecting it should also involve fixing all tests that use it to either capture the warnings or not use the function so the work is likely to be a bit more involved

**Pull request merged**